src/overlays/actors/ovl_En_Owl/z_en_owl.c

```diff
@@ -31,141 +79,1332 @@ const ActorInit En_Owl_InitVars = {
+            break;
+        case OWL_DEATH_MOUNTAIN2:
+            this->actionFunc = EnOwl_WaitDeathMountainShortcut;
+            break;
+        case OWL_DESSERT_COLOSSUS:
+            this->actionFunc = func_80ACB3E0;
+            break;
+        case OWL_LOST_WOODS_PRESARIA:
+            if (!CHECK_QUEST_ITEM(QUEST_SONG_LULLABY)) {
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+            this->actionFunc = EnOwl_WaitLWPreSaria;
+            break;
+        case OWL_LOST_WOODS_POSTSARIA:
+            if (!CHECK_QUEST_ITEM(QUEST_SONG_SARIA)) {
+                osSyncPrintf("フクロウ退避\n"); // Owl evacuation
+                Actor_Kill(&this->actor);
+                return;
+            }
+            this->actionFunc = EnOwl_WaitLWPostSaria;
+            break;
+        default:
+            // Outside kokiri forest
+            osSyncPrintf(VT_FGCOL(CYAN));
+            osSyncPrintf("no = %d  \n", owlType);
+            osSyncPrintf(
+                "未完成のフクロウ未完成のフクロウ未完成のフクロウ\n"); // Unfinished owl unfinished owl unfinished owl
+            osSyncPrintf(VT_RST);
+            this->actionFlags |= 2;
+            this->unk_3EE = 0x20;
+            this->actionFunc = EnOwl_WaitOutsideKokiri;
+            break;
+    }
+}
+
+void EnOwl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOwl* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Owl/func_80ACC5CC.s")
+/*
+ * Rotates this to the player instance
+ */
+void EnOwl_LookAtLink(EnOwl* this, GlobalContext* globalCtx) {
+    s16 yaw;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Owl/func_80ACC624.s")
+    yaw = Math_Vec3f_Yaw(&this->actor.posRot.pos, &player->actor.posRot.pos);
+    this->actor.posRot.rot.y = yaw;
+    this->actor.shape.rot.y = yaw;
+}
 
+/*
+ * Checks if link is within `targetDist` units, initalize the camera for the owl.
+ * returns 0 if the link is not within `targetDistance`, returns 1 once link is within
+ * the distance, and the camera has been initalized.
+ */
+s32 EnOwl_CheckInitTalk(EnOwl* this, GlobalContext* globalCtx, u16 textId, f32 targetDist, u16 flags) {
+    s32 cameraFlags;
+    f32 distCheck;
+
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        if (this->actor.params == 0xFFF) {
+            this->actionFlags |= 0x40;
+            cameraFlags = -0x64;
+        } else {
+            if (Rand_ZeroOne() < 0.5f) {
+                cameraFlags = (flags & 1) != 0 ? -0x61 : -0x63;
+                this->actionFlags |= 0x40;
+            } else {
+                cameraFlags = (flags & 1) ? -0x60 : -0x62;
+                this->actionFlags &= ~0x40;
+            }
+        }
+        this->cameraIdx = func_800800F8(globalCtx, 0x21FC, cameraFlags, &this->actor, 0);
```

Comment:
I thought the third arg for that function was supposed to be a timer or something but I could be wrong.